presto-main/src/test/java/com/facebook/presto/type/TestArrayOperators.java

```diff
@@ -160,6 +163,32 @@ public void testArrayElements()
         assertInvalidFunction("CAST(ARRAY ['puppies', 'kittens'] AS ARRAY<BIGINT>)", INVALID_CAST_ARGUMENT);
     }
 
+    private static String generateLongSring(int size, char c)
+    {
+        char[] cs = new char[size];
+        Arrays.fill(cs, c);
+
+        return new String(cs);
+    }
+
+    private static String generateThreeElementArrayString(int size)
+    {
+        String arrayDistinctSqlString = "array_distinct(ARRAY[" +
+                "'" + generateLongSring(size, 'x') + "', '" +
+                generateLongSring(size, 'y') + "', '" + generateLongSring(size, 'z') + "'])";
+
+        return arrayDistinctSqlString;
+    }
+
+    @SuppressWarnings("NumericCastThatLosesPrecision")
+    @Test
+    public void testArraySize()
+            throws Exception
+    {
+        String sqlString = generateThreeElementArrayString((int) MAX_FUNCTION_MEMORY.toBytes() + 1);
```

Comment:
"Use Math.toIntExact instead of Ints.checkedCast (170:0)"
i got this checkstyle error...